src/update-helm-chart-deps.mjs

```diff
@@ -0,0 +1,138 @@
+        )
+        continue
+      }
+
+      console.log(`Checking updates for dependency: ${dependency.name}`)
+
+      // Add the Helm repository (idempotent)
+      await $`helm repo add ${dependency.name} ${dependency.repository}`
+      await $`helm repo update`
+
+      // Get all available versions for the dependency
+      const allVersions = await $`helm search repo ${dependency.name}/${dependency.name} -o json`
+        .then((output) => JSON.parse(output.stdout))
+        .then((results) => results.map((entry) => entry.version).filter((version) => semver.valid(version)))
+
+      if (!allVersions.length) {
+        console.error(`No valid versions found for dependency ${dependency.name}`)
+        continue
+      }
+
+      // Filter versions for allowed upgrades (minor/patch)
+      const currentVersion = dependency.version
+      const filteredVersions = allVersions.filter((version) => {
+        if (semver.lte(version, currentVersion)) {
+          return false // Ignore versions that are <= current version
+        }
+        if (allowedUpgradeType === 'patch') {
+          return semver.diff(currentVersion, version) === 'patch'
+        }
+        if (allowedUpgradeType === 'minor') {
+          const isMinorOrPatch =
+            semver.diff(currentVersion, version) === 'patch' || semver.diff(currentVersion, version) === 'minor'
+          return isMinorOrPatch
+        }
+        return true // Default: Allow all upgrades
+      })
+
+      if (!filteredVersions.length) {
+        console.log(`No matching ${allowedUpgradeType} updates for dependency ${dependency.name}`)
+        continue
+      }
+
+      // Determine the latest matching version
+      const latestVersion = filteredVersions.sort(semver.rcompare)[0]
+
+      if (latestVersion === currentVersion) {
+        console.log(`${dependency.name} is already up to date.`)
+        continue
+      }
+
+      console.log(`Updating ${dependency.name} from version ${currentVersion} to ${latestVersion}`)
+
+      // Update the version in Chart.yaml
+      dependency.version = latestVersion
+      const branchName = `ci-update-${dependency.name}-to-${latestVersion}`
+      const commitMessage = `chore(chart-deps): update ${dependency.name} to version ${latestVersion}`
+      if (ciCreateFeatureBranch) {
+        // Create a new branch for the update
+        await $`git checkout -b ${branchName}`
+      }
+
+      // Write the updated Chart.yaml file
+      const updatedChart = yaml.dump(chart)
+      await fs.writeFile(chartFile, updatedChart, 'utf8')
+      // Fetch and unpack the new chart version
+      const tempDir = `./tmp/charts/${dependency.name}`
+      await $`mkdir -p ${tempDir}`
+      await $`helm pull ${dependency.name}/${dependency.name} --version ${latestVersion} --destination ${tempDir}`
+      await $`tar -xzvf ${tempDir}/${dependency.name}-${latestVersion}.tgz -C ${chartsDir}`
```

Comment:
I would suggest to add a clean up before, otherwise files renamed or removed in the newer version stay around:
```suggestion
      await $`rm -R ${chartsDir}/${dependency.name}`
      await $`tar -xzvf ${tempDir}/${dependency.name}-${latestVersion}.tgz -C ${chartsDir}`
```

Reply:
Good point!